lib/condor.py

```diff
@@ -54,7 +54,29 @@ def get_schedd(vargs: Dict[str, Any]) -> classad.ClassAd:
     if vargs.get("verbose", 0) > 1:
         print(f"post-query schedd classads: {schedds} ")
 
-    res = random.choice(schedds)
+        print("")
+
+    weighted_schedds = []
+    for s in schedds:
+        name = s.eval("Name")
+        rdcdc = s.eval("RecentDaemonCoreDutyCycle")
+
+        if vargs.get("verbose", 0) > 0:
+            print(f"RecentDaemonCoreDutyCycle {name} {rdcdc}")
+
+        # add to weighted list more times the lower the duty cycle..
+        # if they're *all* not busy, that will put them all in the
+        # list 5 times, but then it will be even odds....
+        weighted_schedds.append(s)
+        if rdcdc < 0.5:
+            weighted_schedds.append(s)
+        if rdcdc < 0.1:
+            weighted_schedds.append(s)
+        if rdcdc < 0.05:
+            weighted_schedds.append(s)
+            weighted_schedds.append(s)
+
+    res = random.choice(weighted_schedds)
```

Comment:
This is fine, but python 3.6 and above has a `random.choices` that allows you to weight your choices.  

https://docs.python.org/3/library/random.html#random.choices

I'll leave it up to you as to whether to use that or not, but it might come out a bit idiomatic/cleaner, especially the part where you compute the weights.

Reply:
Trying that...

Reply:
Does this look better now @shreyb ?